lib/secrets/secrets.go

```diff
@@ -156,18 +179,37 @@ func (d *diskStore) Exists(secret SecretName) (bool, error) {
 }
 
 func (d *diskStore) GetSecret(secret SecretName) ([]byte, error) {
-	path := filepath.Join(d.basePath, secret.Group, secret.Name)
-	ret, err := ioutil.ReadFile(path)
+	var ret []byte
+
+	// checks if the encrypted secret exists
+	pathToEncryptedSecret := filepath.Join(d.basePath, secret.Group, secret.Name+".enc.toml")
+	encryptedSecretExists, err := fileutils.PathExists(pathToEncryptedSecret)
+
+	if encryptedSecretExists {
+		ret, err = ioutil.ReadFile(pathToEncryptedSecret)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not read secrets data (secret may not have been generated yet)")
```

Comment:
This error could be misleading. The file is being checked for existence before coming into the block

Reply:
changed it to be more relatable